File: handlers/firebaseDatabaseHandler.go
package handlers

import (
    "net/http"
    "ur-admin-backend/services"
    "log"
)

func FirebaseDataHandler(w http.ResponseWriter, r *http.Request) {
    if err := CheckLogin(r); err != nil {
        http.Error(w, err.Error(), http.StatusUnauthorized)   
        return
    }

    tokenName := r.URL.Path[len("/api/firebase-data/"):]

    log.Printf("=== FirebaseDataHandler: tokenName: %s", tokenName)

    data, err := services.GetFirebaseData(tokenName)
    if err != nil {
        http.Error(w, "{\"code\":\"500\",\"error\":\""+err.Error()+"\"}", http.StatusInternalServerError)
        return
    }

    response := "{\"code\":200,\"value\":\""+data+"\"}"

    w.WriteHeader(http.StatusOK)
    w.Write([]byte(response))
}

File: handlers/firebaseDatabaseHandler.js
const { getFirebaseData } = require('../services/realtimeDataSvc');
const { checkLogin } = require('./loginHandler');

async function firebaseDataHandler(req, res) {
  try {
    await checkLogin(req);

    const tokenName = req.params.tokenName;

    console.log(`=== FirebaseDataHandler: tokenName: ${tokenName}`);

    const data = await getFirebaseData(tokenName);

    res.status(200).json({ code: 200, value: data });
  } catch (error) {
    if (error.message.includes('Authorization') || error.message.includes('Token') || error.message.includes('Application')) {
      return res.status(401).send(error.message);
    }
    console.error('Firebase Data Error:', error);
    res.status(500).json({ code: "500", error: error.message });
  }
}

module.exports = { firebaseDataHandler };
